core/arch/x86/instr.c

```diff
@@ -2027,6 +2027,184 @@ opnd_same_sizes_ok(opnd_size_t s1, opnd_size_t s2, bool is_reg)
     return (s1_default == s2_default);
 }
 
+opnd_size_t
+instr_get_input_size_from_opcode(int opcode, bool evexw)
+{
+    /* See Intel Vol.2A 2.6.5 Compressed Displacement (disp8*N) Support in EVEX. When it's
+     * not possible determine the input size based on evex.W, we use the opcode. */
+    switch (opcode) {
+    case OP_vpextrb:
+    case OP_vpinsrb:
+    case OP_vpbroadcastb: return OPSZ_1;
+    case OP_vpinsrw:
+    case OP_vpextrw:
+    case OP_vpbroadcastw: return OPSZ_2;
+    case OP_vcvtss2si:
+    case OP_vcvttss2si:
+    case OP_vcomiss:
+    case OP_vcvtss2usi:
+    case OP_vcvttss2usi: return OPSZ_4;
+    case OP_vcvtsd2si:
+    case OP_vcvttsd2si:
+    case OP_vcomisd:
+    case OP_vcvtsd2usi:
+    case OP_vcvttsd2usi: return OPSZ_8;
```

Comment:
Can we make this part of the tuple type?  (Haven't looked in detail but it seems weird to need per-encoding data yet have this separate thing that's also per-opcode.)

Reply:
You probably could. I wanted to keep it more straight forwardly according to how Intel specified it. In case there was a bug, there is less work to do to track it down. If I pulled this into the tuple type, I had to invent new tuple types. I wouldn't want to reverse engineer this later. Can we keep it like this so it is matching Intel (still 2.6.5 Vol. 2A)?
Resolving this, re-open if necessary.